### PR TITLE
feat: welcome window displays what's new

### PR DESCRIPTION
**Pull request by erikas-taroza**
The welcome window will parse the changelog file and display the changes for the current version.

### CODE REVIEW

File: Assets/Mirage/Editor/WelcomeWindow/WelcomeWindow.cs [hunk 1]
```diff
@@ -229,6 +238,72 @@ private void ToggleMenuButtonColor(Button button, bool toggle)
...
+                while((line = reader.ReadLine()) != null)
+                {
+                    //we dont need to parse an empty string
+                    if(line == string.Empty) { continue; }
+
+                    //always add the first line
+                    if(content.Count == 0) { content.Add(line); continue; }
+
+                    //if we havent reached the next version yet
+                    if(!line.Contains("https://github.com/MirageNet/Mirage/compare/"))
```
Comment (paulpach) [inline Assets/Mirage/Editor/WelcomeWindow/WelcomeWindow.cs:259]: a better way:

if the line starts with 
```
# 
```
or
```
## 
```

then we have a new version

File: Assets/Mirage/Editor/WelcomeWindow/Resources/WelcomeWindow.uss [hunk 2]
```diff
@@ -108,17 +108,17 @@ Button {
 }
 
 .dark-selected-tab {
-    background-color: rgba(53, 53, 53, 255);
```
Comment (paulpach) [inline Assets/Mirage/Editor/WelcomeWindow/Resources/WelcomeWindow.uss:111]: I don't think this has anything to do with changelog.  Should be a separate pr

Comment (erikas-taroza) [reply Assets/Mirage/Editor/WelcomeWindow/Resources/WelcomeWindow.uss:111]: Didn't see that. I think Unity automatically changed it for some reason. Will remove this and move it to a different PR later.

File: Assets/Mirage/Editor/WelcomeWindow/WelcomeWindow.cs [hunk 3]
```diff
@@ -229,6 +238,72 @@ private void ToggleMenuButtonColor(Button button, bool toggle)
...
+                    {
+                        content.Add(line);
+                    }
+                    else
+                    {
+                        //break because the next version was reached
+                        break;
+                    }
```
Comment (paulpach) [inline Assets/Mirage/Editor/WelcomeWindow/WelcomeWindow.cs:267]:
```suggestion
                    if(line.Contains("https://github.com/MirageNet/Mirage/compare/"))
                        break;

                    content.Add(line);
```

File: Assets/Mirage/Editor/WelcomeWindow/WelcomeWindow.cs [hunk 4]
```diff
@@ -229,6 +238,67 @@ private void ToggleMenuButtonColor(Button button, bool toggle)
...
+            List<string> content = new List<string>();
+
+            using(StreamReader reader = new StreamReader(changeLogPath))
+            {
+                string line;
+
+                while((line = reader.ReadLine()) != null)
+                {
+                    //we dont need to parse an empty string
+                    if(line == string.Empty) { continue; }
```
Comment (paulpach) [inline Assets/Mirage/Editor/WelcomeWindow/WelcomeWindow.cs:253]:
```suggestion
                    if(line == string.Empty) 
                        continue;
```

File: Assets/Mirage/Editor/WelcomeWindow/WelcomeWindow.cs [hunk 5]
```diff
@@ -229,6 +238,67 @@ private void ToggleMenuButtonColor(Button button, bool toggle)
...
+
+                    //always add the first line
+                    if(content.Count == 0) { content.Add(line); continue; }
```
Comment (paulpach) [inline Assets/Mirage/Editor/WelcomeWindow/WelcomeWindow.cs:256]: one statement per line so that I can use breakpoints

```suggestion
                    if(content.Count == 0) 
                    {
                        content.Add(line); 
                        continue;
                    }
```

File: Assets/Mirage/Editor/WelcomeWindow/WelcomeWindow.cs [hunk 6]
```diff
@@ -58,6 +60,8 @@ public class WelcomeWindow : EditorWindow
         private static string firstStartUpKey = string.Empty;
         private const string firstTimeMirageKey = "MirageWelcome";
 
+        private const string changeLogPath = "Packages/com.miragenet.mirage/CHANGELOG.md";
```
Comment (paulpach) [inline Assets/Mirage/Editor/WelcomeWindow/WelcomeWindow.cs:63]: hardcoded path is an issue,  this does not work if you import it directly without UPM.

What we could do is write the latest version changelog to a file in the resources folder instead.

**Comment by erikas-taroza** [comment]
Will work on this when I have more time.
